clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/ActiveActiveStoreIngestionTask.java

```diff
@@ -615,7 +620,16 @@ private void producePutOrDeleteToKafka(
     GenericRecord rmdRecord = mergeConflictResult.getRmdRecord();
     final ByteBuffer updatedRmdBytes =
         rmdSerDe.serializeRmdRecord(mergeConflictResult.getValueSchemaId(), mergeConflictResult.getRmdRecord());
-
+    // Since data is not chunked in RT but chunked in VT, creating the key for the small record case or CVM to be
+    // used to persist on disk after producing to Kafka.
+    final byte[] updatedKeyBytes =
+        isChunked ? ChunkingUtils.KEY_WITH_CHUNKING_SUFFIX_SERIALIZER.serializeNonChunkedKey(key) : key;
+    final ChunkedValueManifest oldValueManifest = isChunked && msgType.equals(MessageType.UPDATE)
+        ? ChunkingUtils.getChunkValueManifestFromStorage(updatedKeyBytes, subPartition, false, storageEngine)
+        : null;
+    final ChunkedValueManifest oldRmdManifest = isChunked && msgType.equals(MessageType.UPDATE)
+        ? ChunkingUtils.getChunkValueManifestFromStorage(updatedKeyBytes, subPartition, true, storageEngine)
+        : null;
```

Comment:
So for AA use case, RMD value will always be retrieved before executing DCR, I think we should reuse the value retrieved previously.
For UPDATE operation, if RMD permits, the old value will be looked up when executing DCR, so there is a high chance the value manifest is retrieved before this function.
Today, `ChunkingUtils` doesn't expose the details of manifest, so I guess we should update the function interface to include a placeholder, so that internally, `ChunkingUtils` can return the manifest if it is chunked.
This will require a refactoring of `ChunkingUtils`, but I think it is worthwhile considering RocksDB lookup is deemed to be an expensive operation.

Reply:
Also based on the comment description, we should only do these lookups when the message type is `UPDATE`, right?
Right now, it will be executed for operation types.

Reply:
Thanks for the comment, looks like I missed it yesterday. Sure, I was kind of hesitant about this change, but you are right, this should save some more look up.

Reply:
But there is one issue - for those value/RMD that's cached in the transient record, we will have to do the RocksDB get then

Reply:
I wonder whether we can cache the manifest in the transient record or not.

Reply:
Updated the PR, but not yet update the transient record part. Will do later.

Reply:
The latest version should trim value/rmd manifest look up for both cached and not-cached code path for AASIT and LFSIT.